app/api/promiseMiddleware.js

```diff
@@ -1,26 +1,45 @@
 /*
 * Redux middleware to handle promises
-* As seen in: https://github.com/caljrimmer/isomorphic-redux-app
+* As seen in: https://github.com/erikras/react-redux-universal-hot-example/blob/master/src/redux/middleware/clientMiddleware.js
+* 
+* This passes all actions with 'promise' to be used by client, which is custom axios instance
+* Functions called by makeTopicRequest and makeUserRequest are working as intended, because they just don't accept 'client' as argument
 */
+<<<<<<< cfc092ce0e278ad34ed88c597dc39be8a5436836
 
 export default function promiseMiddleware() {
   return next => action => {
     const { promise, type, ...rest } = action;
+=======
+export default function promiseMiddleware(client) {
+  return ({dispatch, getState}) => {
+    return next => action => {
+      if (typeof action === 'function') {
+        return action(dispatch, getState);
```

Comment:
Lol i had no idea redux-thunk was such a [small lib](https://github.com/gaearon/redux-thunk/blob/master/src/index.js#L4). 

I wish we could still use thunk separately from this promise middleware so that we are clear with the differences of each middleware's role.